$NetBSD: patch-ui_qt_CMakeLists.txt,v 1.1 2021/04/06 10:25:01 tnn Exp $

From 6ef6e8f607456e239e5f1a326803846930d34fa8 Mon Sep 17 00:00:00 2001
From: Gerald Combs <gerald@wireshark.org>
Date: Fri, 26 Mar 2021 12:52:01 -0700
Subject: [PATCH] CMake: Enable AUTO{MOC,UIC,RCC} earlier.

Enable CMAKE_AUTOMOC, CMAKE_AUTOUIC, and CMAKE_AUTORCC before searching
for Qt packages. This is apparently required for CMake 3.20.0 and later.
Fixes #17314.

(cherry picked from commit be2b0fc810fb2f2a80e471138dc29f0cbe87b209)

--- ui/qt/CMakeLists.txt.orig	2021-03-10 18:21:20.000000000 +0000
+++ ui/qt/CMakeLists.txt
@@ -631,10 +631,6 @@ include_directories(
 	${CMAKE_CURRENT_SOURCE_DIR}
 )
 
-set(CMAKE_AUTOMOC TRUE)
-set(CMAKE_AUTOUIC TRUE)
-set(CMAKE_AUTORCC TRUE)
-
 if(NOT Qt5Widgets_VERSION VERSION_LESS "5.9")
 	# Drop the file modification time of source files from generated files
 	# to help with reproducible builds. We do not use QFileInfo.lastModified
